### PR TITLE
cifuzz: use pull_request.number instead of GITHUB_REF

### PR DESCRIPTION
**Pull request by evverx**
Closes https://github.com/google/oss-fuzz/issues/3732

### CODE REVIEW

**Comment by evverx** [comment]
To make sure it works I switched to "evverx/oss-fuzz" in https://github.com/systemd/systemd/pull/15820. According https://github.com/systemd/systemd/pull/15820/checks?check_run_id=687355387, that job was successfully restarted via the "re-run jobs" button by checking out "refs/pull/15820/merge".

@jonathanmetzman could you take a look?

File: infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py [hunk 1]
```diff
@@ -40,7 +41,6 @@ def main():
     OSS_FUZZ_PROJECT_NAME: The name of OSS-Fuzz project.
     GITHUB_REPOSITORY: The name of the Github repo that called this script.
     GITHUB_SHA: The commit SHA that triggered this script.
-    GITHUB_REF: The pull request reference that triggered this script.
```
Comment (jonathanmetzman) [inline infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py:43]: Actually, could you describe `GITHUB_EVENT_PATH` here?

**Comment by evverx** [comment]
TravisBuddy is sure annoying :-) Just out of curiosity, I wonder what the point of TravisBuddy is :-)  I'm pretty people can see all and click on those red crosses.

File: infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py (hunk 1 above)
Comment (evverx) [reply infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py:43]: Will do

**Comment by jonathanmetzman** [comment]
> TravisBuddy is sure annoying :-)

Agree 100%. I've been outvoted though on it's usefulness :-(

>Just out of curiosity, I wonder what the point of TravisBuddy is :-) I'm pretty sure people can see and click on all those red crosses.

I think because a lot pull requests to this repo are from casual contributors it helps because:
1. A lot of them aren't as familiar with the workflow of clicking through to see the failures, so it makes the failures more obvious for them.
2. It lets us as maintainers know when someone has fixed their PR and it is ready to land.

File: infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py (hunk 1 above)
Comment (evverx) [reply infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py:43]: Done

**Comment by evverx** [comment]
Got it. Though I think it would be more helpful if it provided direct links to the build logs I can click on instead of comments with "TravisBuddy Request Identifiers" just saying that something is wrong.

**Pull request merged**